fdb-record-layer-jmh/src/jmh/java/com/apple/foundationdb/record/benchmark/lucene/MailBenchmarkUtils.java

```diff
@@ -0,0 +1,154 @@
+import com.apple.foundationdb.record.RecordMetaDataBuilder;
+import com.apple.foundationdb.record.TestMailProto;
+import com.apple.foundationdb.record.lucene.LuceneFunctionNames;
+import com.apple.foundationdb.record.lucene.LuceneIndexTypes;
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.metadata.IndexTypes;
+import com.apple.foundationdb.record.metadata.JoinedRecordTypeBuilder;
+import com.apple.foundationdb.record.metadata.Key;
+import com.apple.foundationdb.record.metadata.RecordTypeBuilder;
+
+import java.time.Duration;
+import java.util.Map;
+
+import static com.apple.foundationdb.record.metadata.Key.Expressions.field;
+import static com.apple.foundationdb.record.metadata.Key.Expressions.function;
+
+/**
+ * Utilities for the javax.mail benchmarks in this package.
+ */
+public class MailBenchmarkUtils {
+
+    static final String msgStateIndex = "msgStateIdx";
+
+    static String toHumanReadableDuration(Duration duration) {
+        return String.format("%d:%d.%d[%d]", duration.toMinutes(), duration.toSeconds(), duration.toMillis(), duration.getNano() / 1000);
+    }
+
+    static RecordMetaData recordStoreMetaData() {
+        RecordMetaDataBuilder rmd = RecordMetaData.newBuilder().setRecords(TestMailProto.getDescriptor());
+        rmd.setSplitLongRecords(true);
+        final RecordTypeBuilder msgStateBuilder = rmd.getRecordType("MessageState");
+        msgStateBuilder.setPrimaryKey(Key.Expressions.field("uid"));
+
+        Index msgStateMboxRefTextSearchRef = new Index(msgStateIndex,
+                Key.Expressions.concat(
+                        Key.Expressions.field("mBoxRef"),
+                        Key.Expressions.field("textSearchRef"),
+                        Key.Expressions.field("uid")),
+                IndexTypes.VALUE
+        );
+        rmd.addIndex("MessageState", msgStateMboxRefTextSearchRef);
+
+
+        Index idx = new Index("fullSearchIndex",
+                Key.Expressions.concat(
+                        function(LuceneFunctionNames.LUCENE_TEXT, field("searchBody")),
+                        function(LuceneFunctionNames.LUCENE_TEXT, field("headerBlob"))
+                ),
+                LuceneIndexTypes.LUCENE,
+                Map.of(
+                        "luceneAnalyzerName", "SYNONYM_MAIL",
+                        "luceneAnalyzerNamePerField", "header_subject:NGRAM,header_from:NGRAM,header_to:NGRAM,header_bcc:NGRAM",
+                        "autoCompleteEnabled", "true",
+                        "autoCompleteExcludedFields", "headerBlob"
+                ));
+
+        rmd.getRecordType("TextSearch").setPrimaryKey(Key.Expressions.field("___recordID"));
+        rmd.addIndex("TextSearch", idx);
+
+        return rmd.build();
+    }
+
+    static RecordMetaData mboxInLuceneMetaData(){
+        RecordMetaDataBuilder rmd = RecordMetaData.newBuilder().setRecords(TestMailProto.getDescriptor());
+        rmd.setSplitLongRecords(true);
```

Comment:
So, metadata can split if we get too big? And lucene index is in the metadata? Could we end up splitting a lucene index?